test/cpp/client/credentials_test.cc

```diff
@@ -667,7 +676,28 @@ TEST_F(CredentialsTest, LoadTlsChannelCredentials) {
       server_authorization_check_config);
   std::shared_ptr<grpc_impl::ChannelCredentials> channel_credentials =
       grpc::experimental::TlsCredentials(options);
-  GPR_ASSERT(channel_credentials != nullptr);
+  GPR_ASSERT(channel_credentials.get() != nullptr);
+}
+
+// This test demonstrates how the TLS credentials will be used to create
+// server credentials.
+TEST_F(CredentialsTest, LoadTlsServerCredentials) {
+  grpc_init();
+  std::shared_ptr<TestTlsCredentialReload> test_credential_reload(
+      new TestTlsCredentialReload());
+  std::shared_ptr<TlsCredentialReloadConfig> credential_reload_config(
+      new TlsCredentialReloadConfig(test_credential_reload));
+
+  // The server builder is needed in order to properly initialize the C-core.
+  ServerBuilder builder;
```

Comment:
It seems this `builder` is not used anywhere. Do you know what part of C-core is initialized due to this `builder`? Also, how is it related to `pthread` errors that we observed?  Sorry for many questions. But I would like to know exactly how this change could fix the `pthread` errors in iOS.

Reply:
This was discussed offline.